### PR TITLE
Backend/mongoose setup

### PR DESCRIPTION
**Pull request by awilli70**
New folder: Models (where we can store/ add schema as necessary)
App.js: Now uses mongoose to perform login. Additionally, has variable to switch between server hosted db and local db

New endpoints using /mongoose-util/:

/getAlumni/
In body: email: String - returns matching alumni/ user

/allAlumni/
Returns JSON object of all alumni in alumni collection

/allUsers/
Returns JSON object of all users in users collection

/addAlumni/
In request body:
        email
        name
        gradYear
        location 
        profession
        company
        college
        availabilities = []; (This can be empty)
        timeZone
        zoomLink
        password

TODO: add requests and posts to alumni profile (Like availabilities these will be empty, but require schema setup)

/data/clear/all, /data/clear/alumni, /data/clear/user
Empties all documents from their respective collections (all empties both)

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -23,9 +24,26 @@ app.use(cookieParser());
...
+ * MongoDB must be installed
+ */
+const testDB = true;
+
+/* Mongoose Setup */
+const mongoose = require('mongoose');
+const uri = testDB ? 'mongodb://localhost:27017/ofi-testdata' : `mongodb://${process.env.DBUSER}:${process.env.DBPASSWORD}@${process.env.DBHOST}/${process.env.DB}`;
+mongoose.connect(uri, {useNewUrlParser: true});
+
+const client = mongoose.connection;
```
Comment (reshadbinharun) [inline server/app.js:39]: Could we move this client instantion in the try/catch of `main()`?

File: server/app.js [hunk 2]
```diff
@@ -23,9 +24,26 @@ app.use(cookieParser());
...
+ * If testDB is true, uses a locally hosted mongoDB
+ * Otherwise, it will use a cloud hosted DB set in the .env file
+ * MongoDB must be installed
+ */
+const testDB = true;
+
+/* Mongoose Setup */
+const mongoose = require('mongoose');
+const uri = testDB ? 'mongodb://localhost:27017/ofi-testdata' : `mongodb://${process.env.DBUSER}:${process.env.DBPASSWORD}@${process.env.DBHOST}/${process.env.DB}`;
+mongoose.connect(uri, {useNewUrlParser: true});
```
Comment (reshadbinharun) [inline server/app.js:37]: Do we need an `await` here? (after we move it into `main())

File: server/app.js [hunk 3]
```diff
@@ -83,18 +98,23 @@ async function main() {
     ));
 
     app.use('/', (req, res, next) => {
-      req.db = db;
+      req.db = client;
```
Comment (reshadbinharun) [inline server/app.js:101]: We would probably not need to pass client on the Request object for each router anymore right?

File: server/routes/utilMongoose.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+var router = express.Router();
+var crypto = require('crypto-random-string');
+var bcrypt = require('bcrypt');
+var mongoose = require('mongoose')
+var userSchema = require('../models/userSchema');
+var alumniSchema = require('../models/alumniSchema');
+
+const HASH_COST = 10;
+
+router.get('/getAlumni/', async (req, res, next) => {
```
Comment (reshadbinharun) [inline server/routes/utilMongoose.js:11]: let's name this path `/alumni` so the `method` is indication of whether it's GET/POST

File: server/routes/utilMongoose.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    try {
+        const dbData = await alumniSchema.find()
+        res.json({'alumnus' : dbData});
+    } catch (e) {
+        console.log("Error: util#allStudents", e);
+        res.status(500).send({'error' : e});
+    }
+});
+
+router.get('/addAlumni/', async (req, res, next) => {
```
Comment (reshadbinharun) [inline server/routes/utilMongoose.js:31]: Should be `router.post`

File: server/routes/utilMongoose.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+              passwordHash: passwordHash,
+              verificationToken: verificationToken,
+              role: role,
+              emailVerified: emailVerified,
+              approved: approved
+            }
+        );
+        
+        alumni_instance.save(function (err) {
+            if (err) return handleError(err);
```
Comment (reshadbinharun) [inline server/routes/utilMongoose.js:79]: is handleError a callback we get with Mongoose?

File: server/routes/utilMongoose.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        res.status(200).send({'message' : 'deleted all records!'});
+    } catch (e) {
+        res.status(500).send({'error' : e});
+    }
+});
+
+router.get('/data/clear/alumni', async (req, res, next) => {
+    try {
+        await alumniSchema.remove({});
+        res.status(200).send({'message' : 'deleted all student records!'});
```
Comment (reshadbinharun) [inline server/routes/utilMongoose.js:117]: nit: alumni* records

File: server/app.js (hunk 1 above)
Comment (awilli70) [reply server/app.js]: Done

File: server/app.js (hunk 2 above)
Comment (awilli70) [reply server/app.js]: Yes- it's been added

File: server/app.js (hunk 3 above)
Comment (awilli70) [reply server/app.js]: Fixed

File: server/routes/utilMongoose.js (hunk 4 above)
Comment (awilli70) [reply server/routes/utilMongoose.js]: Changed

**Pull request merged**